app/models/account.rb

```diff
@@ -1,9 +1,11 @@
 class Account < ApplicationRecord
   belongs_to :user
-  has_many :loyalty_program_datum, class_name: "loyalty_program_data", foreign_key: "reference_id"
+  has_many :loyalty_program_datum, class_name: 'loyalty_program_data', foreign_key: 'reference_id'
   after_create :create_loyalty_program_data
 
   def create_loyalty_program_data
     # Account.
+    byebug
+    @loyalty_program_data = user.account.build_loyalty_program_datum(points: 0).save
```

Comment:
this does not work. need help! @mattChrisP @suvasan19

Reply:
Currently I am not too sure about this as well